estimators/all_estimators.py

```diff
@@ -0,0 +1,279 @@
+# coding=utf-8
+# Copyright 2022 The Google Research Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# """all_estimators!
+#
+# It hadnles the causal inference estimators.
+#
+# """
+
+import itertools
+import logging
+import numpy as np
+import numpy as np
+from sklearn import linear_model
+from sklearn import metrics
+from sklearn import model_selection
+import tensorflow as tf
+
+logger = logging.getLogger(__name__)
+
+
+def estimator_aipw(data, param_method, seed=0):
+    """Estimate treatment effect using AIPW.
+
+    :param data: DataSimulation Class.
+    :param param_method: dict with method's parameters.
+    :param seed: model seed.
+
+    :return:
+    t_estimated: estimated treatment effect using the oaxaca-blinder method,
+    metric: list [,], metric on control and treated groups,
+    bias: list [,], bias on control and treated group,
+    var: float, estimator variance.
+    """
+    # Fix numpy seed for reproducibility
+    np.random.seed(seed)
+    logger.debug('Running AIPW.')
+    # Fitting two models, one per treatment value.
+    logger.debug('Base Model - Control.')
+    y_control_hat, mse_control, e_control, t, y_obs0 = _fit_base_models(model=param_method['base_model'],
+                                                                        data=data.dataset_control,
+                                                                        data_all=data.dataset_all,
+                                                                        param_method=param_method)
+    logger.debug('Base Model - Treated')
+    y_treated_hat, mse_treated, e_treated, t_, y_obs1 = _fit_base_models(model=param_method['base_model'],
+                                                                         data=data.dataset_treated,
+                                                                         data_all=data.dataset_all,
+                                                                         param_method=param_method)
+    assert sum(t) == sum(t_), 'These two values should be identical!'
+    assert sum(y_obs1) == sum(y_obs0), 'These two values should be identical!'
```

Comment:
_fit_base_models() returns the predictions and the observed y. 
There are two _fit_base_models() (one for control, and other for treated). 

This should be true because the data.dataset_all is inputted to both _fit_base_models(), so the observed y should be the same.